views/pages/HomePage.jsx

```diff
@@ -1,27 +1,35 @@
-import React, { useState } from 'react'
+import React, { useEffect, useState } from 'react'
 import { useOutletContext } from 'react-router-dom'
-import ReactSelect from 'react-select'
 
 import ParticipantsTable from '../components/VirtualTables/ParticipantsTable'
-import { components } from '../forms/ControlledReactSelect/components'
-import { useEffect } from 'react'
 import api from '../api'
 import { SORT_DIRECTION } from '../../constants'
+import ParticipantsSearchForm from '../forms/ParticipantsSearchForm'
+import { Box, Typography } from '@mui/material'
 
 const HomePage = () => {
   const { user, setNotification, setUser } = useOutletContext()
   const { uid, preferences } = user
 
+  const [initialLoad, setInitialLoad] = useState(true)
   const [participants, setParticipants] = useState([])
   const [searchSubjects, setSearchSubjects] = useState([])
   const [searchOptions, setSearchOptions] = useState([])
   const [sortDirection, setDirection] = useState(SORT_DIRECTION.ASC)
   const [sortBy, setSortBy] = useState('')
 
-  const fetchParticipants = async (sortParams) =>
-    await api.participants.all(sortParams)
+  const fetchParticipants = React.useCallback(async () => {
```

Comment:
why do we have to useCallback..?

Reply:
this will memoize the function so we can use it as a dependency in the useEffect. Otherwise the function is re-created at every render, causing the useEffect to fire at every render.